resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala

```diff
@@ -51,6 +54,29 @@ private[spark] object KubernetesClientUtils extends Logging {
     propertiesWriter.toString
   }
 
+  object StringLengthOrdering extends Ordering[(String, String)] {
+    override def compare(x: (String, String), y: (String, String)): Int = {
+      // compare based on file length and break the tie with string comparison of keys.
+      (x._1.length + x._2.length).compare(y._1.length + y._2.length) * 10 +
+        x._1.compareTo(y._1)
+    }
```

Comment:
Is this `*10 + ...` correctly?
- https://www.scala-lang.org/api/2.12.10/scala/math/Ordering.html#compare(x:T,y:T):Int

Reply:
Yeah don't you want to compare the sum and only compare the first one if they're equal? do you really even need compare()?

Reply:
Good questions, and hoping that I have understood correctly, attempting to answer.

1) We need to compare the file sizes _including their names_, because that is how they will occupy space in a config map. 
2) We would like to give priority to as many config files as possible we can mount (or store in the configMap), so we would like to sort them by size.
3) If the two files are equal in lengths, then we do not want to declare them equal, so in the compare equation we add their string compare results as well. This is done to break the tie. So `*10` is done to give more priority to compare by their length but if the comparison result is equal, i.e. two files are exactly equal in length then we  get zero as the comparison result. By adding string comparison result of their names, i.e. when two files have equal lengths, we check if the two files have same name as well ? Then we say they are truly equal - in principle this should not happen, because they are files in the same directory i.e. `SPARK_CONF_DIR`. So, each file will get some ordering value as a result of computing comparison equation, and no data will be lost.